src/functions/__tests__/schema-path.test.ts

```diff
@@ -65,6 +69,56 @@ describe('schema', () => {
     ]);
   });
 
+  test('will error with invalid falsy input', () => {
+    const target = {
+      schema: {
+        type: 'string',
+      },
+      example: null,
+    };
+    expect(runSchemaPath(target, fieldToCheck, path)).toEqual([
+      {
+        path: ['example'],
+        message: '{{property|gravis|append-property|optional-typeof}}type should be string',
```

Comment:
What is this syntax? Does type need a space before it?

Reply:
It's replaced at runtime, when Spectral processes messages returned by functions.
`|` acts kind of like the unix pipe.
So in this case, this is what happens:
- `property` prints the property name
- `gravis` is a function that decorates a property with gravises (`\`property\``)
- `append-property` adds `\`${property} property\`` if there is a property or return empty string if there is no property
- optional-typeof prints a type of linted value if `append-property` returned an empty string or the result of append-property.

I know this sounds weird, but so perhaps a note in documentation would be nice to have?

What user receives is something like
```
message: '`example` property type should be number',
```
or
```
message: `number should be equal to one of the allowed values: 1, 3, 5, 10, 12`,
```

That does make sense?

Reply:
If this is something users should know about then please let's document it.

Reply:
This also confused me a lot when working on my recent PRs....

Reply:
> If this is something users should know about then please let's document it.

Will do!

> This also confused me a lot when working on my recent PRs....

My apologies! Going to document that syntax this week.